packages/components/src/components/tabs/tab.jsx

```diff
@@ -3,59 +3,53 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import Counter from '../counter';
 
-class Tab extends React.Component {
-    onClick = () => {
-        const { onClick } = this.props;
-        onClick();
-    };
+const Tab = ({
+    active_tab_ref,
+    bottom,
+    count,
+    header_content,
+    header_fit_content,
+    id,
+    is_active,
+    is_label_hidden,
+    label,
+    onClick,
+    setActiveLineStyle,
+    top,
+}) => {
+    React.useEffect(() => {
+        setActiveLineStyle();
+    }, [count, label, header_content]);
```

Comment:
```suggestion
    const prev_count = usePreviousState(count);
    const prev_label = usePreviousState(label);
    const prev_header_content = usePreviousState(header_content);
    React.useEffect(() => {
        if (
            prev_count !== count ||
            prev_label !== label ||
            prev_header_content !== header_content
        ) setActiveLineStyle();
    }, [count, label, header_content]);
```

Reply:
Salar I think we don't need these statements. the useEffect hook compare the prev_state,  if we had and(&&) in our statements I think we should compare it with the previous state